folly/portability/Windows.h

```diff
@@ -37,6 +37,7 @@
 #include <direct.h> // nolint
 #endif
 
+#define NOMINMAX
```

Comment:
Those can all be moved to portability/Windows.h easily and it'd be the right fix I think. If you think otherwise feel free to close this pull request. If you agree I can update the request moving rest of the defines.